### ISSUE TITLE
[Bug] Audio in Chrome and Safari constant static

### ISSUE THREAD

**Issue by thelamer** [emulatorjs/emulatorjs#739]
I was notified of modification made in https://github.com/libretro/RetroArch/pull/15479 downstream for my docker container and was particularly interested in rwebaudio and openal. I took a stab at modernizing the cores I have with the latest Retroarch upstream and noticed heavy static in the audio, bounced here and found the same thing. If I go to a commit before this merge I will not claim the audio is perfect it still pops and crack from time to time depending on load but the static is gone. 

Here are some videos of current Chromium: (emulatorjs)

https://github.com/EmulatorJS/EmulatorJS/assets/1852688/825a54c1-03f7-4141-b4ec-3551752dc311

https://github.com/EmulatorJS/EmulatorJS/assets/1852688/b7361645-812d-42a1-8479-7f86f834d846

And pre merge build: (libretro core)

https://github.com/EmulatorJS/EmulatorJS/assets/1852688/635367a2-17b2-46e6-9783-92ca0d723f2f

This effects chromium on all platforms it looks like, just tested Windows, Linux, and Xbox Series Edge browser.
This also effects Safari (only have an iPad to test on)
This does not effect Firefox, Firefox audio is vastly improved since the merge. 

What I did notice is that tweaking the baked in output rates and latency in retroarch has a large impact, by default the sampling rate is 48000 while 44100 sounds significantly better with a latency of 64ms. But it still has pops and crackles. I started playing with it after seeing this issue https://github.com/defold/defold/pull/2443. 
![audio_settings](https://github.com/EmulatorJS/EmulatorJS/assets/1852688/ddd338f3-daf0-43b2-a3fa-b6568b5a54fe)

I suspect there may be a disconnect between what libretro is outputting and what emscripten is setting with openal. 

You are generally much lower level than me was hoping you might have some ideas as to what is going on here.

**Comment by thelamer** [comment]
Ok I found the sweet spot here, for safari and chromium swaping to lowest and 44100 produces crystal clear audio. 

I'll make a PR upstream. 

Relevant lines: 
https://github.com/libretro/RetroArch/blob/master/config.def.h#L1103-L1104 (swap to 44100)
https://github.com/libretro/RetroArch/blob/master/config.def.h#L1613 (swap to lowest)

Need new EMSCRIPTEN if guards. 

Here is Chromium now: 


https://github.com/EmulatorJS/EmulatorJS/assets/1852688/8d8a62b4-8b91-43ee-84f6-d6673ca39794

**Comment by ethanaobrien** [comment]
I hadnt even noticed this issue until you brought it up. Thanks for reporting it and making a fix!


Heres the story of why I switched from rwebaudio to openal:

rweb audio faked sleeping, by blocking the main thread (the only thread) until it got to where it was playing audio. This caused many browsers (all of them) to randomly freeze the entire page permanently at start due to a race condition in receiving audio. During the 4.0 rewrite I managed to make the rwebaudio driver stable(..?) for chromium browsers (I do not remember what I changed) but this caused safari on iOS (for some reason iPads were not affected) to completely reload the page upon emulator start.

So basically
- Rwebaudio used blocking functions when js only has one thread, resulting in high cpu usage and crackly audio
- The patched rwebaudio caused safari on iOS devices to completely reload the page
- openal uses basically emscriptens version of rwebaudio, just better

See #416

**Comment by ethanaobrien** [comment]
Fixed in latest release, thank you @thelamer!

**Comment by programmingPug** [comment]
Exact same thing is happening now in FireFox

**Comment by ethanaobrien** [comment]
@programmingPug If you can reproduce using https://demo.emulatorjs.org/, open a bug. Otherwise it isn't my issue

**Comment by programmingPug** [comment]
Just tried and I can, will record a video and open a bug :)
